src/MICore/LaunchOptions.cs

```diff
@@ -109,6 +112,11 @@ static internal PipeLaunchOptions CreateFromXml(Xml.LaunchOptions.PipeLaunchOpti
         /// [Optional] Current working directory when the pipe program is invoked.
         /// </summary>
         public string PipeCwd { get; private set; }
+        
+        /// <summary>
+        /// [Optional] Enviroment variables for the pipe program.
+        /// </summary>
+        public EnvironmentEntry[] PipeEnvironment { get; private set; }
```

Comment:
Can we make this a ReadOnlyCollection and/or IReadOnlyList. I don't like mutable public members of public types